sdk/src/types/block/address/bech32.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::str::FromStr;
+use std::hash::Hash;
+
+use crate::types::block::address::Address;
+
+/// An address and its network type.
+#[derive(Clone, Eq, PartialEq, Hash)]
+pub struct Bech32Address {
+    pub(crate) inner: Address,
+    pub(crate) hrp: String,
+}
+
+impl AsRef<Address> for Bech32Address {
+    fn as_ref(&self) -> &Address {
+        &self.inner
+    }
+}
+
+impl FromStr for Bech32Address {
+    type Err = crate::types::block::Error;
+
+    fn from_str(address: &str) -> Result<Self, Self::Err> {
+        let (address, hrp) = Address::try_from_bech32_with_hrp(address)?;
+
+        Ok(Self::new(address, hrp))
+    }
+}
+
+impl Bech32Address {
+    /// Create a new address wrapper.
+    pub fn new(address: Address, hrp: String) -> Self {
+        Self { inner: address, hrp }
+    }
+
+    /// Get the address part
+    pub fn address(&self) -> &Address {
```

Comment:
Do we want to call it inner?

Reply:
I didnt like that naming, but fine either way.

Reply:
Yeah I don't like it either, but I also don't like address.address, idk